src/spline2/MultiBsplineEval_helper.hpp

```diff
@@ -52,8 +52,13 @@ inline void getSplineBound(T x, TRESIDUAL& dx, int& ind, int nmax)
   }
   else
   {
+#if defined(__INTEL_LLVM_COMPILER) || defined(__INTEL_CLANG_COMPILER)
```

Comment:
Wish that there was some way we could make sure we remove this when the supported intel compilers no longer have broken std::modf.

Reply:
I guess one can add a check the intel compiler version until the broken feature is solved

Reply:
@PDoakORNL since the difference between these two code paths are so small. I doubt it is worth our time checking if intel fixed it from time to time. More likely we will only investigate the issue once the workaround stops working.